df.history.xml

```diff
@@ -3101,32 +3102,28 @@
         <stl-vector name='attacker_civ' type-name='int32_t' ref-target='historical_entity'/>
         <stl-vector name='defender_civ' type-name='int32_t' ref-target='historical_entity'/>
         <stl-vector name='attacker_hf' type-name='int32_t' ref-target='historical_figure'/>
+        <stl-vector name='attacker_role' type-name='int32_t' comment="Tentatively 0: regular, 1/2 merc"/>
         <stl-vector name='defender_hf' type-name='int32_t' ref-target='historical_figure'/>
-        <stl-vector name='noncombat_hf' type-name='int32_t' ref-target='historical_figure'/>
+        <stl-vector name='defender_role' type-name='int32_t' comment="same as for attacker role, i.e. 0-2, with 1/2 being mercs"/>
+        <stl-vector name='noncombat_hf' type-name='int32_t' ref-target='historical_figure' comment="saw being beheaded, but that's only one checked"/>
 
-        these new fields are probably mixed in above or below
-        <stl-vector type-name='int32_t'/>
-        <stl-vector type-name='int32_t'/>
-        <int32_t/>
-        <int32_t init-value='-1'/>
-        <int32_t init-value='-1'/>
+        <bool name='attacker_merc_scouts' comment="else 'ordinary' mercs"/>
+        <int32_t name='attacker_mercs' ref-target='historical_entity'/>
+        <int32_t name='defender_mercs' ref-target='historical_entity'/>
         <stl-vector type-name='int32_t' comment="Observed size of 4 doesn't allow for pointer"/>
         <stl-vector type-name='int32_t' comment="Observed size of 4 doesn't allow for pointer"/>
-        <stl-vector type-name='int32_t'/>
-        <stl-vector type-name='int32_t'/>
-
-        <stl-vector name='attacker_squad_entity_pop' type-name='int32_t'/>
+        <stl-vector name='attacker_squad_entity_pop' type-name='int32_t' ref-target='entity_population'/>
         <stl-vector name='attacker_squad_counts' type-name='int32_t'/>
         <stl-vector name='attacker_squad_deaths' type-name='int32_t'/>
         <stl-vector name='attacker_squad_races' type-name='int32_t' ref-target='creature_raw'/>
         <stl-vector name='attacker_squad_sites' type-name='int32_t' ref-target='world_site'/>
-
-        <stl-vector name='defender_squad_entity_pops' type-name='int32_t'/>
+        <stl-vector type-name='int32_t' comment="probably a boolean, as only 0/1 seen"/>
```

Comment:
(The item type is probably not `bool`, since `std::vector<bool>` has a completely different layout)